File: java/src/main/java/org/urlshortener/core/dtos/UserDto.java
package org.urlshortener.core.dtos;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NonNull;
import org.urlshortener.dal.entities.UserRole;

@Data
@AllArgsConstructor
public class UserDto {
    @NonNull
    private String name;
    @NonNull
    private UserRole role;
    @NonNull
    private String password;
    private Long id;
}

File: kotlin/src/main/kotlin/com/urlshortener/api/dtos/UserDto.kt
package com.urlshortener.api.dtos

import com.urlshortener.dal.entities.UserRole
import com.urlshortener.util.annotations.NoArgs
import javax.validation.constraints.NotNull

@NoArgs
data class CreateUserDto(
    @field:NotNull
    var name: String,
    @field:NotNull
    var role: UserRole,
    @field:NotNull
    var password: String
)

@NoArgs
data class LoginDto(
    @field:NotNull
    var username: String,
    @field:NotNull
    var password: String
)